### ISSUE TITLE
Authentication issue

### ISSUE THREAD

**Issue by minhnimble** [SeintApril/Survey#13]
Currently, after logging in the application successfully and then quit the app, opening the application again will prompt the user to login and the previous session is not persisted at all. I believe there is a requirement from the code challenge to implement the OAuth authentication including the storage of access token with refreshing token capability to persist the session. Are you aware of this requirement and intend to implement it?

**Comment by SeintApril** [comment]
Yes. i'm aware of this requirement but i couldn't implemented it in given time.  The following are how i'm going to implement this feature. 

- call login API
- take the refresh token and access token from response
- save it in the keychain and reuse it whenever i need to call the APIs
- use refresh token to get a new access token when the previous one expired
- replace the old token with the new token which i got from requesting the login API

**Comment by minhnimble** [comment]
It is very nice to see you can describe the actions needed for the refreshing token process. In reality, there are actually a few issues with this flow that requires certain techniques/tools to deal with such as: how to determine when the token expires, how to retry the current failed request(s) because of the expired tokens, etc. Can you tell me more about the techniques to handle the refresh token flow properly?

**Comment by SeintApril** [comment]
Yes. i'm aware of this. The following are how i'm going to implement.
- if current time(unix format) is greater than created_at + expire_in, will request new token.
- will add this condition whenever before calling API except login API.
- For this process, I am not sure which library should I use.

**Comment by minhnimble** [comment]
Thank you for the reply. This is probably all I need to know at this point 🙏.
